jcl/src/java.base/share/classes/java/lang/invoke/VarHandle.java

```diff
@@ -1196,8 +1205,41 @@ public String toString() {
 		 * @return ClassDesc of variable type
 		 */
 		public ClassDesc varType() {
-			return constantType();
+			return (type == Kind.ARRAY) ? getArrayTypeClassDesc().componentType() : getFieldTypeClassDesc();
+		}
+
+		/* helper method. declaring class is always the first bootstrap argument for a field type
+		 * @throws InternalError if VarHandleDesc was created improperly
+		 */
+		private ClassDesc getFieldDeclaringClassDesc() {
+			ConstantDesc[] args = bootstrapArgs();
+			if ((args.length < 2) || !(args[0] instanceof ClassDesc)) {
+				throw new InternalError();
```

Comment:
Please add a description to this exception so that it will be easier to understand what went wrong form a stacktrace